CapstoneProject/Assets/Scripts/Core/Controller/Dialogue/DialogueManager.cs

```diff
@@ -197,33 +213,177 @@ private void BeginText()
     {
         displayTextObject.SetActive(true);
         displayTextObject.GetComponent<TextMeshProUGUI>().enabled = true;
-        StartCoroutine(TextScroll("This is test dialogue. Wa ba ba go bo"));
+        //scrollingCoroutine = TextScrollInput("This is test dialogue.");
+        //StartCoroutine(scrollingCoroutine);
 
+    }
+
+    private IEnumerator MainDialogueLoop()
+    {
+        dialogueIsplaying = true;
+        
+        IEnumerator textCoroutine = null;
         
+        UnityEngine.Debug.Log("Main dialogue loop entered");
+
+
+
+
+        foreach(string dialogue in loadedDialogue)
+        {
+            dialogueTicker++;
+            UnityEngine.Debug.Log("Looping");
+
+            if (gameState.relationshipStatus == 2)
+            {
+                StartCoroutine(SearchForState(textCoroutine, "playerkillrepeat", "break"));
+                dialogueSkipRepeat = true;
+            }
+            else if (gameState.relationshipStatus == 1)
+            {
+                StartCoroutine(SearchForState(textCoroutine, "playerkill", "playerdeathrepeat"));
+                dialogueSkipRepeat = true;
+            }
+            else if (gameState.recentDeath > 2)
+            {
+                StartCoroutine(SearchForState(textCoroutine, "playerkillrepeat", "break"));
+                dialogueSkipRepeat = true;
+            }
+            else if (gameState.recentDeath == 1)
```

Comment:
you could also consider adding a recentDeathRequirement short to the scriptable object and use that for the comparison to load here instead of writing if statements for each specific case of different dialogues.

Reply:
same for the relationship status requirement

Reply:
Doing this would require the game state manager to interface and change the scriptable objects on load before they were used by the dialogue manager rather than the game state manager simply communicating with the dialogue manager; implementation as-is seems fine